src/addons/matchrecursive.js

```diff
@@ -179,10 +180,19 @@ export default (XRegExp) => {
                         break;
                     }
                 }
-            } else {
+            } else if (unbalancedDelimiters === 'error') {
                 const delimSide = rightMatch ? 'right' : 'left';
                 const errorPos = rightMatch ? delimStart : outerStart;
                 throw new Error(`Unbalanced ${delimSide} delimiter found in string at position ${errorPos}`);
+            } else if (unbalancedDelimiters === 'text') {
+                if (rightMatch) {
+                    rightMatch = null;
+                } else {
+                    delimEnd = XRegExp.exec(str, left, outerStart)[0].length + outerStart;
+                    openTokens = 0;
+                }
+            } else {
+                throw new Error(`Unsupported value for unbalancedDelimiters: ${unbalancedDelimiters}`);
```

Comment:
I opted for the design choice of a string "enum" so that there could be room for various other modes in the future ("error left", "error right", "warning", "trim", etc.)

Reply:
Great design choice.